main/autogen/autoloader.cc

```diff
@@ -482,7 +467,10 @@ void populateAutoloadTasksAndCreateDirectories(const core::GlobalState &gs, vect
                                                const AutoloaderConfig &alCfg, string_view path, const DefTree &node) {
     string name = node.root() ? "root" : node.name().show(gs);
     string filePath = join(path, fmt::format("{}.rb", name));
-    tasks.emplace_back(RenderAutoloadTask{filePath, node});
+
+    if (node.mustRender(gs, filePath)) {
```

Comment:
Use `mustRender` to figure out whether we want to schedule an autoload-rendering task for the node.